apps/docs/src/index.data.ts

```diff
@@ -0,0 +1,245 @@
+        ],
+      },
+      {
+        id: "expansion-4",
+        label: "Expansion 4",
+        description: "Low priority components.",
+        components: [
+          { name: "Breadcrumb", implemented: isImplemented("OnyxBreadcrumb") },
+          { name: "Table of Content", implemented: isImplemented("OnyxTableOfContent") },
+          { name: "Wizard", implemented: isImplemented("OnyxWizard") },
+          { name: "Tabs", implemented: isImplemented("OnyxTabs") },
+        ],
+      },
+    ];
+
+    return {
+      componentCount: watchedFiles.length,
+      variantCount,
+      mergedPRCount,
+      closedIssueCount,
+      contributorCount,
+      timestamp: timestamp.toUTCString(),
+      downloads,
+      packageCount: packageFolders.length,
+      componentTabs,
+    };
+  },
+});
+
+/**
+ * Counts the occurrences of the word in the given content.
+ */
+const countWord = (content: string, word: string): number => {
+  return content.split(word).length - 1;
+};
+
+/**
+ * Executes the given GitHub search query and returns the total_count of matched items.
+ *
+ * @see: https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#about-search
+ */
+const searchGitHub = async (
+  endpoint: "issues" | "commits",
+  filterString: string,
+): Promise<number> => {
+  // see: https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#constructing-a-search-query
+  const queryString = encodeURIComponent(`repo:SchwarzIT/onyx ${filterString}`);
+
+  // since we only need the total_count, we can decrease the per_page to 1 to improve request speeds
+  const response = await fetch(
+    `https://api.github.com/search/${endpoint}?q=${queryString}&per_page=1`,
+    {
+      headers: {
+        "X-GitHub-Api-Version": "2022-11-28",
+      },
+    },
+  );
+  const body = await response.json();
+
+  if (response.status < 200 || response.status >= 300) {
+    throw new Error(`GitHub request failed. Response body: ${JSON.stringify(body)}`);
+  }
+
+  if (typeof body !== "object" || typeof body.total_count !== "number") {
+    throw new Error("GitHub search response does not contain total_count");
+  }
+
+  return body.total_count;
+};
+
+/**
+ * Gets the number of contributors.
+ *
+ * @see: https://docs.github.com/en/rest/metrics/statistics?apiVersion=2022-11-28#get-all-contributor-commit-activity
+ */
+const getGitHubContributorCount = async (): Promise<number> => {
+  // since we only need the total_count, we can decrease the per_page to 1 to improve request speeds
+  const response = await fetch("https://api.github.com/repos/SchwarzIT/onyx/stats/contributors", {
+    headers: {
+      "X-GitHub-Api-Version": "2022-11-28",
+    },
+  });
+  const body = await response.json();
+
+  if (response.status < 200 || response.status >= 300) {
+    throw new Error(`GitHub request failed. Response body: ${JSON.stringify(body)}`);
+  }
+
+  if (!Array.isArray(body)) {
+    throw new Error("GitHub contributors data is not an array");
```

Comment:
![image](https://github.com/SchwarzIT/onyx/assets/151019977/972a6225-4be9-452d-b9ff-d55f4b36dfb8)

sometimes I get this error when I run the docs app :(

if it's unstable (only in dev??) maybe could we add an env flag to disable the github count so that not the whole app crashes for development?

Reply:
The GitHub API has a rate limit, depending on how often we restart/change something we might get blocked for 1 minute 😆

Reply:
I changed it to only fetch the data on build

Reply:
thank you!